OpenRA.Mods.Common/Traits/Air/Aircraft.cs

```diff
@@ -604,9 +599,54 @@ public void RemovedFromWorld(Actor self)
 			self.World.ActorMap.RemoveInfluence(self, this);
 			self.World.ActorMap.RemovePosition(self, this);
 			self.World.ScreenMap.Remove(self);
-			IsAirborne = false;
+			OnAirborneAltitudeLeft();
+			OnCruisingAltitudeLeft();
+		}
+
+		#region Airborne upgrades
+
+		void OnAirborneAltitudeReached()
+		{
+			if (airborne || um == null)
```

Comment:
Either replace ` || um == null` with `if (um != null)` above `foreach (var u in Info.AirborneUpgrades)` or rename `airborne` to something like `airborneUpgraded` since `airborne` implies a generic state rather than an upgrade state.